sdk/FunctionMetadataLoaderExtension/Startup.cs

```diff
@@ -50,42 +54,99 @@ public void Configure(IWebJobsBuilder builder)
-                    if (!workerDescription.TryGetValue(_exePathPropertyName, out JToken exePathToken))
-                    {
-                        throw new InvalidOperationException($"The property '{_exePathPropertyName}' is required in '{fullPathToWorkerConfig}'.");
-                    }
+                    throw new FileNotFoundException($"The file '{exePath}' was not found.");
+                }
+            }
+
+            workerDescription.DefaultExecutablePath = exePath;
+        }
+
+        private static WorkerConfigDescription GetWorkerConfigDescription(string workerConfigPath)
+        {
+            if (!File.Exists(workerConfigPath))
+            {
+                throw new FileNotFoundException($"The file '{workerConfigPath}' was not found.");
+            }
+
+            WorkerConfigDescription workerDescription;
+
+            using (var fs = File.OpenText(workerConfigPath))
+            using (var js = new JsonTextReader(fs))
+            {
+                JObject workerDescriptionJObject = (JObject)JToken.ReadFrom(js)["description"];
+                workerDescription = workerDescriptionJObject.ToObject<WorkerConfigDescription>();
 
-                    exePathString = Path.Combine(appRootPath, exePathToken.ToString());
+                if (workerDescription is null)
+                {
+                    throw new InvalidOperationException($"The property 'description' is required in '{workerConfigPath}'.");
                 }
             }
 
-            if (string.IsNullOrEmpty(exePathString))
+            return workerDescription;
+        }
+
+        private static bool HasWorkerRootToken(string exe)
+        {
+            return exe.Contains(WorkerRootToken);
+        }
+
+        private static string AddWorkerRootPath(string exe, string appRootPath)
+        {
+            string execName = exe.Replace(WorkerRootToken, string.Empty);
+
+            if (string.IsNullOrEmpty(execName))
             {
-                throw new InvalidOperationException($"The property '{_exePathPropertyName}' in '{fullPathToWorkerConfig}' cannot be null or empty.");
+                throw new InvalidOperationException($"The property '{ExePathPropertyName}' in '{WorkerConfigFile}' does not contain the executable file name.");
             }
 
-            if (!File.Exists(exePathString))
+            string execPath = Path.Combine(appRootPath, execName);
+
+            return AddExeExtensionIfWindows(execPath);
+        }
+
+        private static string AddExeExtensionIfWindows(string file)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && !file.EndsWith(".exe"))
             {
-                throw new FileNotFoundException($"The file '{exePathString}' was not found.");
+                file += ".exe";
             }
 
-            return exePathString;
+            return file;
+        }
+
+        private class WorkerConfigDescription
```

Comment:
So I had created this when I was starting to update both properties "defaultExcutablePath" and "defaultWorkerPath". But with switching to ".dll" for "defaultWorkerPath" as discussed offline for self-contained, this is no longer required.

I could remove this type and go back to single string. But it doesn't hurt to have this specific type. Thoughts?

Reply:
yeah just keep it.